client/task_runner.go

```diff
@@ -617,14 +619,15 @@ func (r *TaskRunner) Update(update *structs.Allocation) {
 }
 
 // Destroy is used to indicate that the task context should be destroyed
-func (r *TaskRunner) Destroy() {
+func (r *TaskRunner) Destroy(reason string) {
```

Comment:
Make this take a `*structs.TaskEvent`. That way we can add additional detail to the event from the call site